File: app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\User;

/**
 * @property int $id
 * @property string $day
 * @property string $time
 * @property int $category_id
 * @property string $title
 * @property bool $is_done
 * @property-read Category $category
 */

class Task extends Model
{
    use HasFactory;

    protected $fillable = ['user_id', 'task_key', 'day', 'time', 'category_id', 'title', 'is_done'];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: weekly_planner_app/lib/models/task.dart
class Task {
  final int id;
  final String title;
  final String? day;
  final String? time;
  final int? categoryId;
  final bool isDone;

  Task({
    required this.id,
    required this.title,
    this.day,
    this.time,
    this.categoryId,
    required this.isDone,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      day: json['day'],
      time: json['time'],
      categoryId: json['category_id'],
      isDone: json['is_done'] == true || json['is_done'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'day': day,
      'time': time,
      'category_id': categoryId,
      'is_done': isDone,
    };
  }
}